contrib/endpoints/src/api_manager/context/service_context.cc

```diff
@@ -76,6 +76,18 @@ ServiceContext::ServiceContext(std::unique_ptr<ApiManagerEnvInterface> env,
 
   service_account_token_.SetAudience(
       auth::ServiceAccountToken::JWT_TOKEN_FOR_FIREBASE, kFirebaseAudience);
+
+  if (config_->server_config() &&
+      !config_->server_config()
+           ->api_check_security_rules_config()
+           .authorization_service_audience()
+           .empty()) {
+    service_account_token_.SetAudience(
+        auth::ServiceAccountToken::JWT_TOKEN_FOR_AUTHORIZATION_SERVICE,
```

Comment:
Yes. aud is a standard JWT claim. My argument is not that the users don't understand JWTs except that having to provide "aud" here is more confusing. Here is the reason why I think it is more confusing. As a firebase rules author, my thought would be that the user is already authenticated since the user has already provided a valid JWT. Now, the rules file will used the JWT claims to authroize the user. But instead of doing just that, we are also forcing the user to think more about what ESP is going to do which is generate more JWTs. I also don't think it is limiting the user from using multiple authroization sources like cloud datastore AND cloud SQL. Actually, we should encourage our users to not have their firebase rules tied to their datastore schema. This is really painful for the user in the long term for both providing the rules and well as schema changes. Instead, the rules should access some endpoint that is executed as a part of app running in say AppEngine. This allows them to use the provided client libraries that allow for better testing and maintenance. Lets discuss more with Wencheng and Wayne.